web-services/query/src/main/java/datawave/webservice/query/runner/QueryExecutorBean.java

```diff
@@ -2026,6 +2026,7 @@ private BaseQueryResponse next(final String id, boolean checkForContentLookup) {
                 close(id);
                 closedQueryCache.add(id); // remember that we auto-closed this query
             } else {
+                close(id);
```

Comment:
It appears that this close may throw an exception in certain cases.  So this close should be wrapped in a try catch and log the error:
```
try {
  close(id);
} catch (Exception ce) {
  log.error(qe, ce);
}
```